pkg/ingester/limiter_test.go

```diff
@@ -71,88 +71,125 @@ func runLimiterMaxFunctionTest(
-			globalLimit:              900,
-			ringReplicationFactor:    3,
-			ringZoneAwarenessEnabled: true,
-			ringIngesterCount:        9,
-			ringZonesCount:           3,
-			shardSize:                0,
-			expectedValue:            300,
+		"zone-awareness disabled, limit is enabled with replication-factor=3 and the shard size > number of ingesters": {
+			globalLimit:           1000,
+			ringReplicationFactor: 3,
+			ringIngesterCount:     10,
+			shardSize:             20,  // Greater than number of ingesters.
+			expectedValue:         300, // (1000 / 10 healthy ingesters) * 3 replication factor
 		},
-		"zone-awareness enabled, limit enabled and the shard size is NOT divisible by number of zones": {
-			globalLimit:              900,
-			ringReplicationFactor:    3,
-			ringZoneAwarenessEnabled: true,
-			ringIngesterCount:        9,
-			ringZonesCount:           3,
-			shardSize:                5,   // Not divisible by number of zones.
-			expectedValue:            450, // (900 / 6) * 3
+
+		"zone-awareness enabled, limit is disabled": {
+			globalLimit:                 0,
+			ringReplicationFactor:       1,
+			ringIngesterCount:           1,
+			ringZoneAwarenessEnabled:    true,
+			ringZonesCount:              3,
+			healthyIngestersInZoneCount: 1,
+			expectedValue:               math.MaxInt32,
 		},
-		"zone-awareness enabled, limit enabled and the shard size is divisible by number of zones": {
-			globalLimit:              900,
-			ringReplicationFactor:    3,
-			ringZoneAwarenessEnabled: true,
-			ringIngesterCount:        9,
-			ringZonesCount:           3,
-			shardSize:                6,   // Divisible by number of zones.
-			expectedValue:            450, // (900 / 6) * 3
+		"zone-awareness enabled, limit is enabled with replication-factor=3, all ingesters up and running, and shard size 0": {
+			globalLimit:                 900,
+			ringReplicationFactor:       3,
+			ringIngesterCount:           9,
+			ringZoneAwarenessEnabled:    true,
+			ringZonesCount:              3,
+			healthyIngestersInZoneCount: 3, // 9 ingesters / 3 zones
+			shardSize:                   0,
+			expectedValue:               300, // (900 / 3 zones / 3 healthy ingesters per zone) * 3 replication factor
 		},
-		"zone-awareness enabled, limit enabled and the shard size > number of ingesters": {
-			globalLimit:              900,
-			ringReplicationFactor:    3,
-			ringZoneAwarenessEnabled: true,
-			ringIngesterCount:        9,
-			ringZonesCount:           3,
-			shardSize:                20, // Greater than number of ingesters.
-			expectedValue:            300,
+		"zone-awareness enabled, limit is enabled with replication-factor=3, some ingesters scaled down, and shard size 0": {
+			globalLimit:                 900,
+			ringReplicationFactor:       3,
+			ringIngesterCount:           9,
+			ringZoneAwarenessEnabled:    true,
+			ringZonesCount:              3,
+			healthyIngestersInZoneCount: 2, // in this zone ingesters are scaled down from 3 to 2
```

Comment:
In that case, total number of ingesters (`ringIngesterCount`) should be 8, not 9.